### ISSUE TITLE
npm i && npm test fails

### ISSUE THREAD

**Issue by jjoekoullas** [badeball/karma-jsdom-launcher#30]
Cloned this repo and ran `npm i && npm test` - tests immediately failed.

```
module.js:549
    throw err;
    ^

Error: Cannot find module 'karma'
```

**Comment by jjoekoullas** [comment]
I see there's a `clean-install` npm script which installs karma & jsdom via the command line.

Is there a particular reason these weren't listed as devDependencies instead?

**Comment by jjoekoullas** [comment]
`npm run clean-install && npm test` fails with

```
added 341 packages in 181.227s
ln: failed to create symbolic link 'node_modules/karma-jsdom-launcher' -> '../': File name too long
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! karma-jsdom-launcher@7.1.0 clean-install: `rm -rf node_modules/ && npm install && npm install --no-save karma@4 jsdom@13 && ln -s ../ node_modules/karma-jsdom-launcher`
```

**Comment by badeball** [comment]
> Is there a particular reason these weren't listed as devDependencies instead?

Yes, but it's no longer entirely clear to me anymore, IE. kind of historical reasons.

NPM has a history of being notorously bad for library developers developing libraries with peer dependencies [1] and Yarn isn't any better [2]. I seem to recall behavior changing somewhat frequently between major versions, while the most common use cases with NPM remains stable.

[1] https://github.com/npm/npm/issues/11213
[2] https://github.com/yarnpkg/yarn/issues/1503

What I'm looking for is a way to install all combinations of peer dependencies, while

1. being easy to use
2. not create noice (ie. not modify package-lock, etc.)

If you (or anyone else) know of a solution that satisfies those requirements, then shout out.

In Ruby-land, I think I would have multiple lock-files, one for each combination. However, I don't think this is possible with NPM.

> `npm run clean-install && npm test` fails with
> 
> ```
> [...]
> File name too long
> ```

Are you running this on Windows? This will probably not work, as I've assumed in `package.json` for `/` to be file separator.
